Use crop hashes to filter out originals from crops

joemcgill opened the pull request:
This is a possible approach to #155, which is a tricky problem since WP core doesn't explicitly describe an image size as a crop vs an original. Feedback would be appreciated.

jaspermdegroot commented on wp-tevko-responsive-images.php:
```diff
@@ -209,9 +209,19 @@ function tevkori_get_srcset_array( $id, $size = 'thumbnail' ) {
+	// If the image has been cropped, it will contain a unique hash.
+	// We look for that hash and use it later to filter out images that aren't
+	//  of the same crop.
+	$img_cropped = preg_match( '/e\d{13}/', $img_url, $img_crop_hash );
```
WP core uses ```'/-e[0-9]{13}\./'``` (wp-admin/includes/image-edit.php) so maybe it's good to use the same regex.

jaspermdegroot commented on wp-tevko-responsive-images.php:
```diff
@@ -209,9 +209,19 @@ function tevkori_get_srcset_array( $id, $size = 'thumbnail' ) {
 
+	// If the image has been cropped, it will contain a unique hash.
+	// We look for that hash and use it later to filter out images that aren't
+	//  of the same crop.
```
Remove extra space at beginning of the line

jaspermdegroot commented on wp-tevko-responsive-images.php:
```diff
@@ -209,9 +209,19 @@ function tevkori_get_srcset_array( $id, $size = 'thumbnail' ) {
 	// Only use sizes with same aspect ratio.
 	foreach ( $img_sizes as $img ) {
 
+		// filter out images that are from different crops
```
Start with capital and end with period.

jaspermdegroot commented:
The problem as described in https://github.com/ResponsiveImagesCG/wp-tevko-responsive-images/issues/155#issuecomment-135862662 does not only occur when you use the crop tool but also when you (for example) scale and flip an image. The code of this PR handles all cases, but maybe we should change comments and variable names to something more generic ("edited" instead of "cropped")

joemcgill commented:
+1 good suggestion.

joemcgill commented:
@jaspermdegroot I updated the logic to filter out non-edits based on your feedback and added tests. I went ahead and merged both since you had already provided feedback, but feel free to look again and let me know if you have any concerns. Thanks.

jaspermdegroot commented:
@joemcgill - This looks good to me. Only one minor thing; one comment was still about crops (see my comment in the diff).

jaspermdegroot commented on wp-tevko-responsive-images.php:
```diff
@@ -209,10 +209,21 @@ function tevkori_get_srcset_array( $id, $size = 'thumbnail' ) {
 	foreach ( $img_sizes as $img ) {
 
-		// Calculate the new image ratio
+		// Filter out images that are from different crops.
```
"Filter out images that are left overs from previous renditions."

joemcgill commented:
Good catch @jaspermdegroot. Fixed in 5ca463e
